app/models/concerns/foreman_teamdynamix/host_managed_extensions.rb

```diff
@@ -0,0 +1,21 @@
+module ForemanTeamdynamix
+  module HostManagedExtensions
+    extend ActiveSupport::Concern
+
+    def td_api
+      @td_api || TeamdynamixApi.new
+    end
+
```

Comment:
You need a validation like this to ensure uniqueness of the teamdynamix_asset_id: https://github.com/theforeman/foreman/blob/develop/app/models/host/managed.rb#L198